aten/src/ATen/native/quantized/cpu/qupsample_bilinear2d.cpp

```diff
@@ -0,0 +1,180 @@
+#include <ATen/ATen.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/cpu/vec256/vec256.h>
+#include <ATen/native/SortingUtils.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/UpSample.h>
+#include <ATen/native/cpu/Loops.h>
+#include <ATen/quantized/Quantizer.h>
+#include <ATen/native/quantized/cpu/quantized_ops.h>
+
+#include <algorithm>
+#include <cmath>
+#include <limits>
+
+namespace at {
+namespace native {
+namespace {
+
+// at::native functions for the native_functions.yaml
+template <typename scalar_t>
+static void upsample_bilinear2d_out_frame(
+    Tensor& output,
+    const Tensor& input,
+    int64_t input_height,
+    int64_t input_width,
+    int64_t output_height,
+    int64_t output_width,
+    int64_t nbatch,
+    int64_t channels,
+    bool align_corners) {
+  auto* idata = static_cast<scalar_t*>(input.data_ptr());
+  auto* odata = static_cast<scalar_t*>(output.data_ptr());
+
+  channels = channels * nbatch;
+  auto* i_p = reinterpret_cast<typename scalar_t::underlying*>(idata);
+  auto* o_p = reinterpret_cast<typename scalar_t::underlying*>(odata);
+
+  // special case: just copy
+  if (input_height == output_height && input_width == output_width) {
+    for (int64_t h2 = 0; h2 < output_height; ++h2) {
+      const int64_t h1 = h2;
+
+      for (int64_t w2 = 0; w2 < output_width; ++w2) {
+        const int64_t w1 = w2;
+        const auto* pos1 = &i_p[h1 * input_width + w1];
+        auto* pos2 = &o_p[h2 * output_width + w2];
+
+        for (int64_t c = 0; c < channels; ++c) {
+          pos2[0] = pos1[0];
```

Comment:
Same comment as the other PR: this section is autovectorizable and should probably go in kernels/

Reply:
I think memcpy makes sense. But it is already a library call, which should always have its own dispatching for AVX/AVX2. So I still think we should keep this part here.

Reply:
`memcpy` can be replaced with a builtin and inlined by the compiler. We should not be handrolling memcpy like this.